Bouwblokken kunnen interne case status niet updaten

## Betrokken versie
13.17

## Verwacht gedrag
Als ik een update zaakstatus en interne status bouwblok wil gebruiken dan lukt het zetten van de interne status niet

## Huidig gedrag
Ik krijg onderstaande error als ik het bouwblok aanroep vanuit een dossier:
`Caused by: org.springframework.orm.jpa.JpaObjectRetrievalFailureException: Unable to find com.ritense.document.domain.InternalCaseStatus with id InternalCaseStatusId(caseDefinitionKey=update-zaakstatus-en-interne-status, key=aanvraag-ontvangen)`

## Mogelijke oplossing (optioneel)
Zorgen dat hij niet kijkt naar de context van het bouwbklok als caseDefinitionKey maar naar de case die het bouwblok aanroept

## Stappen om te reproduceren
*Geef een duidelijke reeks stappen om deze bug te reproduceren.*
*Voeg code toe om te reproduceren, indien relevant*
1. Maak een case aan
2. Maak een bouwblok dat de interne status updatet
3. Koppel de call activity in de case aan het bouwblok
4. Voer het proces uit

--- a/backend/app/gzac/src/dev/resources/config/building-block/household-verification/1-0-0/bpmn/household-verification-process.bpmn
+++ b/backend/app/gzac/src/dev/resources/config/building-block/household-verification/1-0-0/bpmn/household-verification-process.bpmn
@@ -5,10 +5,18 @@
   </bpmn:collaboration>
   <bpmn:process id="household-verification-process" name="Huishouden Verificatie Process" isExecutable="true" camunda:historyTimeToLive="180">
     <bpmn:startEvent id="StartEvent" name="Start Verificatie">
-      <bpmn:outgoing>Flow_Start_To_Format</bpmn:outgoing>
+      <bpmn:outgoing>Flow_Start_To_SetStatusVerificatie</bpmn:outgoing>
     </bpmn:startEvent>
+    <bpmn:serviceTask id="SetStatusVerificatieTask" name="Status: Verificatie Gestart" camunda:asyncAfter="true" camunda:expression="${documentDelegateService.setInternalStatus(execution, 'verificatie-gestart')}">
+      <bpmn:incoming>Flow_Start_To_SetStatusVerificatie</bpmn:incoming>
+      <bpmn:outgoing>Flow_SetStatusVerificatie_To_AddTag</bpmn:outgoing>
+    </bpmn:serviceTask>
+    <bpmn:serviceTask id="AddTagVerificatieTask" name="Tag: Verificatie Vereist" camunda:asyncAfter="true" camunda:expression="${documentDelegateService.addCaseTag(execution, 'verificatie-vereist')}">
+      <bpmn:incoming>Flow_SetStatusVerificatie_To_AddTag</bpmn:incoming>
+      <bpmn:outgoing>Flow_AddTag_To_Format</bpmn:outgoing>
+    </bpmn:serviceTask>
     <bpmn:serviceTask id="FormatNameTask" name="Formatteer Naam" camunda:asyncAfter="true" camunda:expression="${valueResolverDelegateService.handleValue(execution, 'doc:/formattedName', valueResolverDelegateService.resolveValue(execution, 'doc:/applicantName').toUpperCase())}">
-      <bpmn:incoming>Flow_Start_To_Format</bpmn:incoming>
+      <bpmn:incoming>Flow_AddTag_To_Format</bpmn:incoming>
       <bpmn:outgoing>Flow_Format_To_BRP</bpmn:outgoing>
     </bpmn:serviceTask>
     <bpmn:callActivity id="CallBrpLookupActivity" name="BRP Opvraging" camunda:asyncAfter="true" calledElement="brp-lookup" camunda:calledElementBinding="versionTag" camunda:calledElementVersionTag="BB:household-verification:1.0.0">
@@ -50,7 +58,9 @@
       <bpmn:incoming>Flow_Calculate_To_End</bpmn:incoming>
     </bpmn:endEvent>
     <bpmn:sequenceFlow id="Flow_Calculate_To_End" sourceRef="CalculateBaseAmountTask" targetRef="EndEvent" />
-    <bpmn:sequenceFlow id="Flow_Start_To_Format" sourceRef="StartEvent" targetRef="FormatNameTask" />
+    <bpmn:sequenceFlow id="Flow_Start_To_SetStatusVerificatie" sourceRef="StartEvent" targetRef="SetStatusVerificatieTask" />
+    <bpmn:sequenceFlow id="Flow_SetStatusVerificatie_To_AddTag" sourceRef="SetStatusVerificatieTask" targetRef="AddTagVerificatieTask" />
+    <bpmn:sequenceFlow id="Flow_AddTag_To_Format" sourceRef="AddTagVerificatieTask" targetRef="FormatNameTask" />
     <bpmn:sequenceFlow id="Flow_Format_To_BRP" sourceRef="FormatNameTask" targetRef="CallBrpLookupActivity" />
     <bpmn:sequenceFlow id="Flow_BRP_To_IncomeCheck" sourceRef="CallBrpLookupActivity" targetRef="CallIncomeCheckActivity" />
     <bpmn:sequenceFlow id="Flow_IncomeCheck_To_Verify" sourceRef="CallIncomeCheckActivity" targetRef="CreateVerifiedApplicantTask" />
@@ -58,7 +68,7 @@
   <bpmndi:BPMNDiagram id="BPMNDiagram_1">
     <bpmndi:BPMNPlane id="BPMNPlane_1" bpmnElement="Collaboration_HouseholdVerification">
       <bpmndi:BPMNShape id="Participant_HouseholdVerification_di" bpmnElement="HouseholdVerificationParticipant" isHorizontal="true">
-        <dc:Bounds x="160" y="60" width="1310" height="250" />
+        <dc:Bounds x="160" y="60" width="1570" height="250" />
         <bpmndi:BPMNLabel />
       </bpmndi:BPMNShape>
       <bpmndi:BPMNShape id="StartEvent_di" bpmnElement="StartEvent">
@@ -67,71 +77,87 @@
           <dc:Bounds x="202" y="202" width="77" height="27" />
         </bpmndi:BPMNLabel>
       </bpmndi:BPMNShape>
-      <bpmndi:BPMNShape id="FormatNameTask_di" bpmnElement="FormatNameTask">
+      <bpmndi:BPMNShape id="SetStatusVerificatieTask_di" bpmnElement="SetStatusVerificatieTask">
         <dc:Bounds x="290" y="137" width="100" height="80" />
         <bpmndi:BPMNLabel />
       </bpmndi:BPMNShape>
-      <bpmndi:BPMNShape id="CallBrpLookupActivity_di" bpmnElement="CallBrpLookupActivity">
+      <bpmndi:BPMNShape id="AddTagVerificatieTask_di" bpmnElement="AddTagVerificatieTask">
         <dc:Bounds x="420" y="137" width="100" height="80" />
         <bpmndi:BPMNLabel />
       </bpmndi:BPMNShape>
-      <bpmndi:BPMNShape id="CallIncomeCheckActivity_di" bpmnElement="CallIncomeCheckActivity">
+      <bpmndi:BPMNShape id="FormatNameTask_di" bpmnElement="FormatNameTask">
         <dc:Bounds x="550" y="137" width="100" height="80" />
         <bpmndi:BPMNLabel />
       </bpmndi:BPMNShape>
-      <bpmndi:BPMNShape id="CreateVerifiedApplicantTask_di" bpmnElement="CreateVerifiedApplicantTask">
+      <bpmndi:BPMNShape id="CallBrpLookupActivity_di" bpmnElement="CallBrpLookupActivity">
         <dc:Bounds x="680" y="137" width="100" height="80" />
         <bpmndi:BPMNLabel />
       </bpmndi:BPMNShape>
-      <bpmndi:BPMNShape id="PatchZaakVerificationTask_di" bpmnElement="PatchZaakVerificationTask">
+      <bpmndi:BPMNShape id="CallIncomeCheckActivity_di" bpmnElement="CallIncomeCheckActivity">
         <dc:Bounds x="810" y="137" width="100" height="80" />
         <bpmndi:BPMNLabel />
       </bpmndi:BPMNShape>
-      <bpmndi:BPMNShape id="ReadZaakToelichtingTask_di" bpmnElement="ReadZaakToelichtingTask">
+      <bpmndi:BPMNShape id="CreateVerifiedApplicantTask_di" bpmnElement="CreateVerifiedApplicantTask">
         <dc:Bounds x="940" y="137" width="100" height="80" />
         <bpmndi:BPMNLabel />
       </bpmndi:BPMNShape>
-      <bpmndi:BPMNShape id="CalculateBaseAmountTask_di" bpmnElement="CalculateBaseAmountTask">
+      <bpmndi:BPMNShape id="PatchZaakVerificationTask_di" bpmnElement="PatchZaakVerificationTask">
         <dc:Bounds x="1070" y="137" width="100" height="80" />
         <bpmndi:BPMNLabel />
       </bpmndi:BPMNShape>
+      <bpmndi:BPMNShape id="ReadZaakToelichtingTask_di" bpmnElement="ReadZaakToelichtingTask">
+        <dc:Bounds x="1200" y="137" width="100" height="80" />
+        <bpmndi:BPMNLabel />
+      </bpmndi:BPMNShape>
+      <bpmndi:BPMNShape id="CalculateBaseAmountTask_di" bpmnElement="CalculateBaseAmountTask">
+        <dc:Bounds x="1330" y="137" width="100" height="80" />
+        <bpmndi:BPMNLabel />
+      </bpmndi:BPMNShape>
       <bpmndi:BPMNShape id="EndEvent_di" bpmnElement="EndEvent">
-        <dc:Bounds x="1232" y="159" width="36" height="36" />
+        <dc:Bounds x="1492" y="159" width="36" height="36" />
         <bpmndi:BPMNLabel>
-          <dc:Bounds x="1210" y="202" width="80" height="27" />
+          <dc:Bounds x="1470" y="202" width="80" height="27" />
         </bpmndi:BPMNLabel>
       </bpmndi:BPMNShape>
-      <bpmndi:BPMNEdge id="Flow_Start_To_Format_di" bpmnElement="Flow_Start_To_Format">
+      <bpmndi:BPMNEdge id="Flow_Start_To_SetStatusVerificatie_di" bpmnElement="Flow_Start_To_SetStatusVerificatie">
         <di:waypoint x="258" y="177" />
         <di:waypoint x="290" y="177" />
       </bpmndi:BPMNEdge>
-      <bpmndi:BPMNEdge id="Flow_Format_To_BRP_di" bpmnElement="Flow_Format_To_BRP">
+      <bpmndi:BPMNEdge id="Flow_SetStatusVerificatie_To_AddTag_di" bpmnElement="Flow_SetStatusVerificatie_To_AddTag">
         <di:waypoint x="390" y="177" />
         <di:waypoint x="420" y="177" />
       </bpmndi:BPMNEdge>
-      <bpmndi:BPMNEdge id="Flow_BRP_To_IncomeCheck_di" bpmnElement="Flow_BRP_To_IncomeCheck">
+      <bpmndi:BPMNEdge id="Flow_AddTag_To_Format_di" bpmnElement="Flow_AddTag_To_Format">
         <di:waypoint x="520" y="177" />
         <di:waypoint x="550" y="177" />
       </bpmndi:BPMNEdge>
-      <bpmndi:BPMNEdge id="Flow_IncomeCheck_To_Verify_di" bpmnElement="Flow_IncomeCheck_To_Verify">
+      <bpmndi:BPMNEdge id="Flow_Format_To_BRP_di" bpmnElement="Flow_Format_To_BRP">
         <di:waypoint x="650" y="177" />
         <di:waypoint x="680" y="177" />
       </bpmndi:BPMNEdge>
-      <bpmndi:BPMNEdge id="Flow_Verify_To_PatchZaak_di" bpmnElement="Flow_Verify_To_PatchZaak">
+      <bpmndi:BPMNEdge id="Flow_BRP_To_IncomeCheck_di" bpmnElement="Flow_BRP_To_IncomeCheck">
         <di:waypoint x="780" y="177" />
         <di:waypoint x="810" y="177" />
       </bpmndi:BPMNEdge>
-      <bpmndi:BPMNEdge id="Flow_PatchZaak_To_ReadToelichting_di" bpmnElement="Flow_PatchZaak_To_ReadToelichting">
+      <bpmndi:BPMNEdge id="Flow_IncomeCheck_To_Verify_di" bpmnElement="Flow_IncomeCheck_To_Verify">
         <di:waypoint x="910" y="177" />
         <di:waypoint x="940" y="177" />
       </bpmndi:BPMNEdge>
-      <bpmndi:BPMNEdge id="Flow_ReadToelichting_To_Calculate_di" bpmnElement="Flow_ReadToelichting_To_Calculate">
+      <bpmndi:BPMNEdge id="Flow_Verify_To_PatchZaak_di" bpmnElement="Flow_Verify_To_PatchZaak">
         <di:waypoint x="1040" y="177" />
         <di:waypoint x="1070" y="177" />
       </bpmndi:BPMNEdge>
-      <bpmndi:BPMNEdge id="Flow_Calculate_To_End_di" bpmnElement="Flow_Calculate_To_End">
+      <bpmndi:BPMNEdge id="Flow_PatchZaak_To_ReadToelichting_di" bpmnElement="Flow_PatchZaak_To_ReadToelichting">
         <di:waypoint x="1170" y="177" />
-        <di:waypoint x="1232" y="177" />
+        <di:waypoint x="1200" y="177" />
+      </bpmndi:BPMNEdge>
+      <bpmndi:BPMNEdge id="Flow_ReadToelichting_To_Calculate_di" bpmnElement="Flow_ReadToelichting_To_Calculate">
+        <di:waypoint x="1300" y="177" />
+        <di:waypoint x="1330" y="177" />
+      </bpmndi:BPMNEdge>
+      <bpmndi:BPMNEdge id="Flow_Calculate_To_End_di" bpmnElement="Flow_Calculate_To_End">
+        <di:waypoint x="1430" y="177" />
+        <di:waypoint x="1492" y="177" />
       </bpmndi:BPMNEdge>
     </bpmndi:BPMNPlane>
   </bpmndi:BPMNDiagram>

--- a/backend/app/gzac/src/dev/resources/config/building-block/income-check/1-0-0/bpmn/income-check-process.bpmn
+++ b/backend/app/gzac/src/dev/resources/config/building-block/income-check/1-0-0/bpmn/income-check-process.bpmn
@@ -5,10 +5,18 @@
   </bpmn:collaboration>
   <bpmn:process id="income-check-process" name="Inkomenstoets Process" isExecutable="true" camunda:historyTimeToLive="180">
     <bpmn:startEvent id="StartEvent" name="Start Inkomenstoets">
-      <bpmn:outgoing>Flow_Start_To_Check</bpmn:outgoing>
+      <bpmn:outgoing>Flow_Start_To_SetStatusInkomenstoets</bpmn:outgoing>
     </bpmn:startEvent>
+    <bpmn:serviceTask id="SetStatusInkomenstoetsTask" name="Status: Inkomenstoets Gestart" camunda:asyncAfter="true" camunda:expression="${documentDelegateService.setInternalStatus(execution, 'inkomenstoets-gestart')}">
+      <bpmn:incoming>Flow_Start_To_SetStatusInkomenstoets</bpmn:incoming>
+      <bpmn:outgoing>Flow_SetStatusInkomenstoets_To_RemoveTagPriority</bpmn:outgoing>
+    </bpmn:serviceTask>
+    <bpmn:serviceTask id="RemoveTagPriorityTask" name="Verwijder Tag: Prioriteit" camunda:asyncAfter="true" camunda:expression="${documentDelegateService.removeCaseTag(execution, 'prioriteit-hoog')}">
+      <bpmn:incoming>Flow_SetStatusInkomenstoets_To_RemoveTagPriority</bpmn:incoming>
+      <bpmn:outgoing>Flow_RemoveTagPriority_To_Check</bpmn:outgoing>
+    </bpmn:serviceTask>
     <bpmn:serviceTask id="DetermineEligibilityTask" name="Bepaal Geschiktheid" camunda:asyncAfter="true" camunda:expression="${valueResolverDelegateService.handleValue(execution, 'doc:/eligibilityResult', 'Geschikt: '.concat(valueResolverDelegateService.resolveValue(execution, 'doc:/applicantName')).concat(' (huishouden: ').concat(valueResolverDelegateService.resolveValue(execution, 'doc:/householdSize')).concat(' personen)'))}">
-      <bpmn:incoming>Flow_Start_To_Check</bpmn:incoming>
+      <bpmn:incoming>Flow_RemoveTagPriority_To_Check</bpmn:incoming>
       <bpmn:outgoing>Flow_Check_To_Status</bpmn:outgoing>
     </bpmn:serviceTask>
     <bpmn:serviceTask id="SetCheckStatusTask" name="Registreer Status" camunda:asyncAfter="true" camunda:expression="${valueResolverDelegateService.handleValue(execution, 'doc:/checkStatus', 'GOEDGEKEURD')}">
@@ -30,7 +38,9 @@
     <bpmn:endEvent id="EndEvent" name="Inkomenstoets Afgerond">
       <bpmn:incoming>Flow_ReadEinddatum_To_End</bpmn:incoming>
     </bpmn:endEvent>
-    <bpmn:sequenceFlow id="Flow_Start_To_Check" sourceRef="StartEvent" targetRef="DetermineEligibilityTask" />
+    <bpmn:sequenceFlow id="Flow_Start_To_SetStatusInkomenstoets" sourceRef="StartEvent" targetRef="SetStatusInkomenstoetsTask" />
+    <bpmn:sequenceFlow id="Flow_SetStatusInkomenstoets_To_RemoveTagPriority" sourceRef="SetStatusInkomenstoetsTask" targetRef="RemoveTagPriorityTask" />
+    <bpmn:sequenceFlow id="Flow_RemoveTagPriority_To_Check" sourceRef="RemoveTagPriorityTask" targetRef="DetermineEligibilityTask" />
     <bpmn:sequenceFlow id="Flow_Check_To_Status" sourceRef="DetermineEligibilityTask" targetRef="SetCheckStatusTask" />
     <bpmn:sequenceFlow id="Flow_Status_To_SetDate" sourceRef="SetCheckStatusTask" targetRef="SetEinddatumGeplandTask" />
     <bpmn:sequenceFlow id="Flow_SetDate_To_PatchZaak" sourceRef="SetEinddatumGeplandTask" targetRef="PatchZaakIncomeCheckTask" />
@@ -40,7 +50,7 @@
   <bpmndi:BPMNDiagram id="BPMNDiagram_1">
     <bpmndi:BPMNPlane id="BPMNPlane_1" bpmnElement="Collaboration_IncomeCheck">
       <bpmndi:BPMNShape id="Participant_IncomeCheck_di" bpmnElement="IncomeCheckParticipant" isHorizontal="true">
-        <dc:Bounds x="160" y="60" width="980" height="200" />
+        <dc:Bounds x="160" y="60" width="1240" height="200" />
         <bpmndi:BPMNLabel />
       </bpmndi:BPMNShape>
       <bpmndi:BPMNShape id="StartEvent_di" bpmnElement="StartEvent">
@@ -49,55 +59,71 @@
           <dc:Bounds x="200" y="185" width="81" height="27" />
         </bpmndi:BPMNLabel>
       </bpmndi:BPMNShape>
-      <bpmndi:BPMNShape id="DetermineEligibilityTask_di" bpmnElement="DetermineEligibilityTask">
+      <bpmndi:BPMNShape id="SetStatusInkomenstoetsTask_di" bpmnElement="SetStatusInkomenstoetsTask">
         <dc:Bounds x="290" y="120" width="100" height="80" />
         <bpmndi:BPMNLabel />
       </bpmndi:BPMNShape>
-      <bpmndi:BPMNShape id="SetCheckStatusTask_di" bpmnElement="SetCheckStatusTask">
+      <bpmndi:BPMNShape id="RemoveTagPriorityTask_di" bpmnElement="RemoveTagPriorityTask">
         <dc:Bounds x="420" y="120" width="100" height="80" />
         <bpmndi:BPMNLabel />
       </bpmndi:BPMNShape>
-      <bpmndi:BPMNShape id="SetEinddatumGeplandTask_di" bpmnElement="SetEinddatumGeplandTask">
+      <bpmndi:BPMNShape id="DetermineEligibilityTask_di" bpmnElement="DetermineEligibilityTask">
         <dc:Bounds x="550" y="120" width="100" height="80" />
         <bpmndi:BPMNLabel />
       </bpmndi:BPMNShape>
-      <bpmndi:BPMNShape id="PatchZaakIncomeCheckTask_di" bpmnElement="PatchZaakIncomeCheckTask">
+      <bpmndi:BPMNShape id="SetCheckStatusTask_di" bpmnElement="SetCheckStatusTask">
         <dc:Bounds x="680" y="120" width="100" height="80" />
         <bpmndi:BPMNLabel />
       </bpmndi:BPMNShape>
-      <bpmndi:BPMNShape id="ReadZaakEinddatumTask_di" bpmnElement="ReadZaakEinddatumTask">
+      <bpmndi:BPMNShape id="SetEinddatumGeplandTask_di" bpmnElement="SetEinddatumGeplandTask">
         <dc:Bounds x="810" y="120" width="100" height="80" />
         <bpmndi:BPMNLabel />
       </bpmndi:BPMNShape>
+      <bpmndi:BPMNShape id="PatchZaakIncomeCheckTask_di" bpmnElement="PatchZaakIncomeCheckTask">
+        <dc:Bounds x="940" y="120" width="100" height="80" />
+        <bpmndi:BPMNLabel />
+      </bpmndi:BPMNShape>
+      <bpmndi:BPMNShape id="ReadZaakEinddatumTask_di" bpmnElement="ReadZaakEinddatumTask">
+        <dc:Bounds x="1070" y="120" width="100" height="80" />
+        <bpmndi:BPMNLabel />
+      </bpmndi:BPMNShape>
       <bpmndi:BPMNShape id="EndEvent_di" bpmnElement="EndEvent">
-        <dc:Bounds x="972" y="142" width="36" height="36" />
+        <dc:Bounds x="1232" y="142" width="36" height="36" />
         <bpmndi:BPMNLabel>
-          <dc:Bounds x="948" y="185" width="85" height="27" />
+          <dc:Bounds x="1208" y="185" width="85" height="27" />
         </bpmndi:BPMNLabel>
       </bpmndi:BPMNShape>
-      <bpmndi:BPMNEdge id="Flow_Start_To_Check_di" bpmnElement="Flow_Start_To_Check">
+      <bpmndi:BPMNEdge id="Flow_Start_To_SetStatusInkomenstoets_di" bpmnElement="Flow_Start_To_SetStatusInkomenstoets">
         <di:waypoint x="258" y="160" />
         <di:waypoint x="290" y="160" />
       </bpmndi:BPMNEdge>
-      <bpmndi:BPMNEdge id="Flow_Check_To_Status_di" bpmnElement="Flow_Check_To_Status">
+      <bpmndi:BPMNEdge id="Flow_SetStatusInkomenstoets_To_RemoveTagPriority_di" bpmnElement="Flow_SetStatusInkomenstoets_To_RemoveTagPriority">
         <di:waypoint x="390" y="160" />
         <di:waypoint x="420" y="160" />
       </bpmndi:BPMNEdge>
-      <bpmndi:BPMNEdge id="Flow_Status_To_SetDate_di" bpmnElement="Flow_Status_To_SetDate">
+      <bpmndi:BPMNEdge id="Flow_RemoveTagPriority_To_Check_di" bpmnElement="Flow_RemoveTagPriority_To_Check">
         <di:waypoint x="520" y="160" />
         <di:waypoint x="550" y="160" />
       </bpmndi:BPMNEdge>
-      <bpmndi:BPMNEdge id="Flow_SetDate_To_PatchZaak_di" bpmnElement="Flow_SetDate_To_PatchZaak">
+      <bpmndi:BPMNEdge id="Flow_Check_To_Status_di" bpmnElement="Flow_Check_To_Status">
         <di:waypoint x="650" y="160" />
         <di:waypoint x="680" y="160" />
       </bpmndi:BPMNEdge>
-      <bpmndi:BPMNEdge id="Flow_PatchZaak_To_ReadEinddatum_di" bpmnElement="Flow_PatchZaak_To_ReadEinddatum">
+      <bpmndi:BPMNEdge id="Flow_Status_To_SetDate_di" bpmnElement="Flow_Status_To_SetDate">
         <di:waypoint x="780" y="160" />
         <di:waypoint x="810" y="160" />
       </bpmndi:BPMNEdge>
-      <bpmndi:BPMNEdge id="Flow_ReadEinddatum_To_End_di" bpmnElement="Flow_ReadEinddatum_To_End">
+      <bpmndi:BPMNEdge id="Flow_SetDate_To_PatchZaak_di" bpmnElement="Flow_SetDate_To_PatchZaak">
         <di:waypoint x="910" y="160" />
-        <di:waypoint x="972" y="160" />
+        <di:waypoint x="940" y="160" />
+      </bpmndi:BPMNEdge>
+      <bpmndi:BPMNEdge id="Flow_PatchZaak_To_ReadEinddatum_di" bpmnElement="Flow_PatchZaak_To_ReadEinddatum">
+        <di:waypoint x="1040" y="160" />
+        <di:waypoint x="1070" y="160" />
+      </bpmndi:BPMNEdge>
+      <bpmndi:BPMNEdge id="Flow_ReadEinddatum_To_End_di" bpmnElement="Flow_ReadEinddatum_To_End">
+        <di:waypoint x="1170" y="160" />
+        <di:waypoint x="1232" y="160" />
       </bpmndi:BPMNEdge>
     </bpmndi:BPMNPlane>
   </bpmndi:BPMNDiagram>

--- a/backend/app/gzac/src/dev/resources/config/building-block/subsidy-calculator/1-0-0/bpmn/subsidy-calculator-process.bpmn
+++ b/backend/app/gzac/src/dev/resources/config/building-block/subsidy-calculator/1-0-0/bpmn/subsidy-calculator-process.bpmn
@@ -5,10 +5,18 @@
   </bpmn:collaboration>
   <bpmn:process id="subsidy-calculator-process" name="Subsidie Berekening Process" isExecutable="true" camunda:historyTimeToLive="180">
     <bpmn:startEvent id="StartEvent" name="Start Berekening">
-      <bpmn:outgoing>Flow_Start_To_Init</bpmn:outgoing>
+      <bpmn:outgoing>Flow_Start_To_SetStatusBerekening</bpmn:outgoing>
     </bpmn:startEvent>
+    <bpmn:serviceTask id="SetStatusBerekeningTask" name="Status: Berekening Gestart" camunda:asyncAfter="true" camunda:expression="${documentDelegateService.setInternalStatus(execution, 'berekening-gestart')}">
+      <bpmn:incoming>Flow_Start_To_SetStatusBerekening</bpmn:incoming>
+      <bpmn:outgoing>Flow_SetStatusBerekening_To_AddTagPriority</bpmn:outgoing>
+    </bpmn:serviceTask>
+    <bpmn:serviceTask id="AddTagPriorityTask" name="Tag: Prioriteit Hoog" camunda:asyncAfter="true" camunda:expression="${documentDelegateService.addCaseTag(execution, 'prioriteit-hoog')}">
+      <bpmn:incoming>Flow_SetStatusBerekening_To_AddTagPriority</bpmn:incoming>
+      <bpmn:outgoing>Flow_AddTagPriority_To_Init</bpmn:outgoing>
+    </bpmn:serviceTask>
     <bpmn:serviceTask id="InitializeCalculationTask" name="Initialiseer Berekening" camunda:asyncAfter="true" camunda:expression="${valueResolverDelegateService.handleValue(execution, 'doc:/calculationStatus', 'GESTART')}">
-      <bpmn:incoming>Flow_Start_To_Init</bpmn:incoming>
+      <bpmn:incoming>Flow_AddTagPriority_To_Init</bpmn:incoming>
       <bpmn:outgoing>Flow_Init_To_PatchZaak</bpmn:outgoing>
     </bpmn:serviceTask>
     <bpmn:serviceTask id="PatchZaakCalculationStartTask" name="Update Zaak: Start Berekening" camunda:asyncAfter="true">
@@ -33,14 +41,38 @@
     </bpmn:serviceTask>
     <bpmn:serviceTask id="CalculateFinalAmountTask" name="Bereken Eindtotaal" camunda:asyncAfter="true" camunda:expression="${valueResolverDelegateService.handleValue(execution, 'doc:/finalSubsidyAmount', valueResolverDelegateService.resolveValue(execution, 'doc:/baseSubsidyAmount') + 100)}">
       <bpmn:incoming>Flow_Summary_To_Calculate</bpmn:incoming>
-      <bpmn:outgoing>Flow_Calculate_To_Review</bpmn:outgoing>
+      <bpmn:outgoing>Flow_Calculate_To_SetAssignee</bpmn:outgoing>
+    </bpmn:serviceTask>
+    <bpmn:serviceTask id="SetAssigneeTask" name="Toewijzen Beoordelaar" camunda:asyncAfter="true" camunda:expression="${documentDelegateService.setAssignee(execution, 'user@example.com')}">
+      <bpmn:incoming>Flow_Calculate_To_SetAssignee</bpmn:incoming>
+      <bpmn:outgoing>Flow_SetAssignee_To_SetStatusReview</bpmn:outgoing>
+    </bpmn:serviceTask>
+    <bpmn:serviceTask id="SetStatusReviewTask" name="Status: In Beoordeling" camunda:asyncAfter="true" camunda:expression="${documentDelegateService.setInternalStatus(execution, 'in-beoordeling')}">
+      <bpmn:incoming>Flow_SetAssignee_To_SetStatusReview</bpmn:incoming>
+      <bpmn:outgoing>Flow_SetStatusReview_To_Review</bpmn:outgoing>
     </bpmn:serviceTask>
     <bpmn:userTask id="ReviewCalculationTask" name="Beoordeel Berekening" camunda:candidateGroups="ROLE_USER">
-      <bpmn:incoming>Flow_Calculate_To_Review</bpmn:incoming>
-      <bpmn:outgoing>Flow_Review_To_Complete</bpmn:outgoing>
+      <bpmn:incoming>Flow_SetStatusReview_To_Review</bpmn:incoming>
+      <bpmn:outgoing>Flow_Review_To_RemoveTagVerificatie</bpmn:outgoing>
     </bpmn:userTask>
+    <bpmn:serviceTask id="RemoveTagVerificatieTask" name="Verwijder Tag: Verificatie" camunda:asyncAfter="true" camunda:expression="${documentDelegateService.removeCaseTag(execution, 'verificatie-vereist')}">
+      <bpmn:incoming>Flow_Review_To_RemoveTagVerificatie</bpmn:incoming>
+      <bpmn:outgoing>Flow_RemoveTagVerificatie_To_AddTagGoedgekeurd</bpmn:outgoing>
+    </bpmn:serviceTask>
+    <bpmn:serviceTask id="AddTagGoedgekeurdTask" name="Tag: Goedgekeurd" camunda:asyncAfter="true" camunda:expression="${documentDelegateService.addCaseTag(execution, 'goedgekeurd')}">
+      <bpmn:incoming>Flow_RemoveTagVerificatie_To_AddTagGoedgekeurd</bpmn:incoming>
+      <bpmn:outgoing>Flow_AddTagGoedgekeurd_To_Unassign</bpmn:outgoing>
+    </bpmn:serviceTask>
+    <bpmn:serviceTask id="UnassignTask" name="Toewijzing Opheffen" camunda:asyncAfter="true" camunda:expression="${documentDelegateService.unassign(execution)}">
+      <bpmn:incoming>Flow_AddTagGoedgekeurd_To_Unassign</bpmn:incoming>
+      <bpmn:outgoing>Flow_Unassign_To_SetStatusAfgehandeld</bpmn:outgoing>
+    </bpmn:serviceTask>
+    <bpmn:serviceTask id="SetStatusAfgehandeldTask" name="Status: Afgehandeld" camunda:asyncAfter="true" camunda:expression="${documentDelegateService.setInternalStatus(execution, 'afgehandeld')}">
+      <bpmn:incoming>Flow_Unassign_To_SetStatusAfgehandeld</bpmn:incoming>
+      <bpmn:outgoing>Flow_SetStatusAfgehandeld_To_Complete</bpmn:outgoing>
+    </bpmn:serviceTask>
     <bpmn:serviceTask id="MarkCalculationCompleteTask" name="Markeer Voltooid" camunda:asyncAfter="true" camunda:expression="${valueResolverDelegateService.handleValue(execution, 'doc:/calculationStatus', 'VOLTOOID')}">
-      <bpmn:incoming>Flow_Review_To_Complete</bpmn:incoming>
+      <bpmn:incoming>Flow_SetStatusAfgehandeld_To_Complete</bpmn:incoming>
       <bpmn:outgoing>Flow_Complete_To_StoreZgwSummary</bpmn:outgoing>
     </bpmn:serviceTask>
     <bpmn:serviceTask id="StoreZgwActionSummaryTask" name="Bewaar ZGW Samenvatting" camunda:asyncAfter="true" camunda:expression="${valueResolverDelegateService.handleValue(execution, 'doc:/zgwActionSummary', 'Omschrijving (BB1): '.concat(valueResolverDelegateService.resolveValue(execution, 'zaak:omschrijving')).concat(' | Toelichting (BB2): ').concat(valueResolverDelegateService.resolveValue(execution, 'zaak:toelichting')).concat(' | Einddatum (BB3): ').concat(valueResolverDelegateService.resolveValue(execution, 'zaak:einddatumGepland')))}">
@@ -48,12 +80,20 @@
       <bpmn:outgoing>Flow_StoreZgwSummary_To_End</bpmn:outgoing>
     </bpmn:serviceTask>
     <bpmn:sequenceFlow id="Flow_Summary_To_Calculate" sourceRef="GenerateSummaryTask" targetRef="CalculateFinalAmountTask" />
-    <bpmn:sequenceFlow id="Flow_Calculate_To_Review" sourceRef="CalculateFinalAmountTask" targetRef="ReviewCalculationTask" />
-    <bpmn:sequenceFlow id="Flow_Review_To_Complete" sourceRef="ReviewCalculationTask" targetRef="MarkCalculationCompleteTask" />
+    <bpmn:sequenceFlow id="Flow_Calculate_To_SetAssignee" sourceRef="CalculateFinalAmountTask" targetRef="SetAssigneeTask" />
+    <bpmn:sequenceFlow id="Flow_SetAssignee_To_SetStatusReview" sourceRef="SetAssigneeTask" targetRef="SetStatusReviewTask" />
+    <bpmn:sequenceFlow id="Flow_SetStatusReview_To_Review" sourceRef="SetStatusReviewTask" targetRef="ReviewCalculationTask" />
+    <bpmn:sequenceFlow id="Flow_Review_To_RemoveTagVerificatie" sourceRef="ReviewCalculationTask" targetRef="RemoveTagVerificatieTask" />
+    <bpmn:sequenceFlow id="Flow_RemoveTagVerificatie_To_AddTagGoedgekeurd" sourceRef="RemoveTagVerificatieTask" targetRef="AddTagGoedgekeurdTask" />
+    <bpmn:sequenceFlow id="Flow_AddTagGoedgekeurd_To_Unassign" sourceRef="AddTagGoedgekeurdTask" targetRef="UnassignTask" />
+    <bpmn:sequenceFlow id="Flow_Unassign_To_SetStatusAfgehandeld" sourceRef="UnassignTask" targetRef="SetStatusAfgehandeldTask" />
+    <bpmn:sequenceFlow id="Flow_SetStatusAfgehandeld_To_Complete" sourceRef="SetStatusAfgehandeldTask" targetRef="MarkCalculationCompleteTask" />
     <bpmn:endEvent id="EndEvent" name="Berekening Afgerond">
       <bpmn:incoming>Flow_StoreZgwSummary_To_End</bpmn:incoming>
     </bpmn:endEvent>
-    <bpmn:sequenceFlow id="Flow_Start_To_Init" sourceRef="StartEvent" targetRef="InitializeCalculationTask" />
+    <bpmn:sequenceFlow id="Flow_Start_To_SetStatusBerekening" sourceRef="StartEvent" targetRef="SetStatusBerekeningTask" />
+    <bpmn:sequenceFlow id="Flow_SetStatusBerekening_To_AddTagPriority" sourceRef="SetStatusBerekeningTask" targetRef="AddTagPriorityTask" />
+    <bpmn:sequenceFlow id="Flow_AddTagPriority_To_Init" sourceRef="AddTagPriorityTask" targetRef="InitializeCalculationTask" />
     <bpmn:sequenceFlow id="Flow_Init_To_PatchZaak" sourceRef="InitializeCalculationTask" targetRef="PatchZaakCalculationStartTask" />
     <bpmn:sequenceFlow id="Flow_PatchZaak_To_ReadToelichting" sourceRef="PatchZaakCalculationStartTask" targetRef="ReadZaakToelichtingTask" />
     <bpmn:sequenceFlow id="Flow_ReadToelichting_To_CallHousehold" sourceRef="ReadZaakToelichtingTask" targetRef="CallHouseholdVerificationActivity" />
@@ -64,7 +104,7 @@
   <bpmndi:BPMNDiagram id="BPMNDiagram_1">
     <bpmndi:BPMNPlane id="BPMNPlane_1" bpmnElement="Collaboration_SubsidyCalculator">
       <bpmndi:BPMNShape id="Participant_SubsidyCalculator_di" bpmnElement="SubsidyCalculatorParticipant" isHorizontal="true">
-        <dc:Bounds x="160" y="60" width="1680" height="250" />
+        <dc:Bounds x="160" y="60" width="2830" height="250" />
         <bpmndi:BPMNLabel />
       </bpmndi:BPMNShape>
       <bpmndi:BPMNShape id="StartEvent_di" bpmnElement="StartEvent">
@@ -73,87 +113,151 @@
           <dc:Bounds x="196" y="202" width="88" height="27" />
         </bpmndi:BPMNLabel>
       </bpmndi:BPMNShape>
-      <bpmndi:BPMNShape id="InitializeCalculationTask_di" bpmnElement="InitializeCalculationTask">
+      <bpmndi:BPMNShape id="SetStatusBerekeningTask_di" bpmnElement="SetStatusBerekeningTask">
         <dc:Bounds x="290" y="137" width="100" height="80" />
         <bpmndi:BPMNLabel />
       </bpmndi:BPMNShape>
-      <bpmndi:BPMNShape id="PatchZaakCalculationStartTask_di" bpmnElement="PatchZaakCalculationStartTask">
+      <bpmndi:BPMNShape id="AddTagPriorityTask_di" bpmnElement="AddTagPriorityTask">
         <dc:Bounds x="420" y="137" width="100" height="80" />
         <bpmndi:BPMNLabel />
       </bpmndi:BPMNShape>
-      <bpmndi:BPMNShape id="ReadZaakToelichtingTask_di" bpmnElement="ReadZaakToelichtingTask">
+      <bpmndi:BPMNShape id="InitializeCalculationTask_di" bpmnElement="InitializeCalculationTask">
         <dc:Bounds x="550" y="137" width="100" height="80" />
         <bpmndi:BPMNLabel />
       </bpmndi:BPMNShape>
-      <bpmndi:BPMNShape id="CallHouseholdVerificationActivity_di" bpmnElement="CallHouseholdVerificationActivity">
+      <bpmndi:BPMNShape id="PatchZaakCalculationStartTask_di" bpmnElement="PatchZaakCalculationStartTask">
         <dc:Bounds x="680" y="137" width="100" height="80" />
         <bpmndi:BPMNLabel />
       </bpmndi:BPMNShape>
-      <bpmndi:BPMNShape id="GenerateSummaryTask_di" bpmnElement="GenerateSummaryTask">
+      <bpmndi:BPMNShape id="ReadZaakToelichtingTask_di" bpmnElement="ReadZaakToelichtingTask">
         <dc:Bounds x="810" y="137" width="100" height="80" />
         <bpmndi:BPMNLabel />
       </bpmndi:BPMNShape>
-      <bpmndi:BPMNShape id="CalculateFinalAmountTask_di" bpmnElement="CalculateFinalAmountTask">
+      <bpmndi:BPMNShape id="CallHouseholdVerificationActivity_di" bpmnElement="CallHouseholdVerificationActivity">
         <dc:Bounds x="940" y="137" width="100" height="80" />
         <bpmndi:BPMNLabel />
       </bpmndi:BPMNShape>
-      <bpmndi:BPMNShape id="ReviewCalculationTask_di" bpmnElement="ReviewCalculationTask">
+      <bpmndi:BPMNShape id="GenerateSummaryTask_di" bpmnElement="GenerateSummaryTask">
         <dc:Bounds x="1070" y="137" width="100" height="80" />
         <bpmndi:BPMNLabel />
       </bpmndi:BPMNShape>
-      <bpmndi:BPMNShape id="MarkCalculationCompleteTask_di" bpmnElement="MarkCalculationCompleteTask">
+      <bpmndi:BPMNShape id="CalculateFinalAmountTask_di" bpmnElement="CalculateFinalAmountTask">
         <dc:Bounds x="1200" y="137" width="100" height="80" />
         <bpmndi:BPMNLabel />
       </bpmndi:BPMNShape>
-      <bpmndi:BPMNShape id="StoreZgwActionSummaryTask_di" bpmnElement="StoreZgwActionSummaryTask">
+      <bpmndi:BPMNShape id="SetAssigneeTask_di" bpmnElement="SetAssigneeTask">
         <dc:Bounds x="1330" y="137" width="100" height="80" />
         <bpmndi:BPMNLabel />
       </bpmndi:BPMNShape>
+      <bpmndi:BPMNShape id="SetStatusReviewTask_di" bpmnElement="SetStatusReviewTask">
+        <dc:Bounds x="1460" y="137" width="100" height="80" />
+        <bpmndi:BPMNLabel />
+      </bpmndi:BPMNShape>
+      <bpmndi:BPMNShape id="ReviewCalculationTask_di" bpmnElement="ReviewCalculationTask">
+        <dc:Bounds x="1590" y="137" width="100" height="80" />
+        <bpmndi:BPMNLabel />
+      </bpmndi:BPMNShape>
+      <bpmndi:BPMNShape id="RemoveTagVerificatieTask_di" bpmnElement="RemoveTagVerificatieTask">
+        <dc:Bounds x="1720" y="137" width="100" height="80" />
+        <bpmndi:BPMNLabel />
+      </bpmndi:BPMNShape>
+      <bpmndi:BPMNShape id="AddTagGoedgekeurdTask_di" bpmnElement="AddTagGoedgekeurdTask">
+        <dc:Bounds x="1850" y="137" width="100" height="80" />
+        <bpmndi:BPMNLabel />
+      </bpmndi:BPMNShape>
+      <bpmndi:BPMNShape id="UnassignTask_di" bpmnElement="UnassignTask">
+        <dc:Bounds x="1980" y="137" width="100" height="80" />
+        <bpmndi:BPMNLabel />
+      </bpmndi:BPMNShape>
+      <bpmndi:BPMNShape id="SetStatusAfgehandeldTask_di" bpmnElement="SetStatusAfgehandeldTask">
+        <dc:Bounds x="2110" y="137" width="100" height="80" />
+        <bpmndi:BPMNLabel />
+      </bpmndi:BPMNShape>
+      <bpmndi:BPMNShape id="MarkCalculationCompleteTask_di" bpmnElement="MarkCalculationCompleteTask">
+        <dc:Bounds x="2240" y="137" width="100" height="80" />
+        <bpmndi:BPMNLabel />
+      </bpmndi:BPMNShape>
+      <bpmndi:BPMNShape id="StoreZgwActionSummaryTask_di" bpmnElement="StoreZgwActionSummaryTask">
+        <dc:Bounds x="2500" y="137" width="100" height="80" />
+        <bpmndi:BPMNLabel />
+      </bpmndi:BPMNShape>
       <bpmndi:BPMNShape id="EndEvent_di" bpmnElement="EndEvent">
-        <dc:Bounds x="1492" y="159" width="36" height="36" />
+        <dc:Bounds x="2662" y="159" width="36" height="36" />
         <bpmndi:BPMNLabel>
-          <dc:Bounds x="1472" y="202" width="77" height="27" />
+          <dc:Bounds x="2642" y="202" width="77" height="27" />
         </bpmndi:BPMNLabel>
       </bpmndi:BPMNShape>
-      <bpmndi:BPMNEdge id="Flow_Start_To_Init_di" bpmnElement="Flow_Start_To_Init">
+      <bpmndi:BPMNEdge id="Flow_Start_To_SetStatusBerekening_di" bpmnElement="Flow_Start_To_SetStatusBerekening">
         <di:waypoint x="258" y="177" />
         <di:waypoint x="290" y="177" />
       </bpmndi:BPMNEdge>
-      <bpmndi:BPMNEdge id="Flow_Init_To_PatchZaak_di" bpmnElement="Flow_Init_To_PatchZaak">
+      <bpmndi:BPMNEdge id="Flow_SetStatusBerekening_To_AddTagPriority_di" bpmnElement="Flow_SetStatusBerekening_To_AddTagPriority">
         <di:waypoint x="390" y="177" />
         <di:waypoint x="420" y="177" />
       </bpmndi:BPMNEdge>
-      <bpmndi:BPMNEdge id="Flow_PatchZaak_To_ReadToelichting_di" bpmnElement="Flow_PatchZaak_To_ReadToelichting">
+      <bpmndi:BPMNEdge id="Flow_AddTagPriority_To_Init_di" bpmnElement="Flow_AddTagPriority_To_Init">
         <di:waypoint x="520" y="177" />
         <di:waypoint x="550" y="177" />
       </bpmndi:BPMNEdge>
-      <bpmndi:BPMNEdge id="Flow_ReadToelichting_To_CallHousehold_di" bpmnElement="Flow_ReadToelichting_To_CallHousehold">
+      <bpmndi:BPMNEdge id="Flow_Init_To_PatchZaak_di" bpmnElement="Flow_Init_To_PatchZaak">
         <di:waypoint x="650" y="177" />
         <di:waypoint x="680" y="177" />
       </bpmndi:BPMNEdge>
-      <bpmndi:BPMNEdge id="Flow_CallHousehold_To_Summary_di" bpmnElement="Flow_CallHousehold_To_Summary">
+      <bpmndi:BPMNEdge id="Flow_PatchZaak_To_ReadToelichting_di" bpmnElement="Flow_PatchZaak_To_ReadToelichting">
         <di:waypoint x="780" y="177" />
         <di:waypoint x="810" y="177" />
       </bpmndi:BPMNEdge>
-      <bpmndi:BPMNEdge id="Flow_Summary_To_Calculate_di" bpmnElement="Flow_Summary_To_Calculate">
+      <bpmndi:BPMNEdge id="Flow_ReadToelichting_To_CallHousehold_di" bpmnElement="Flow_ReadToelichting_To_CallHousehold">
         <di:waypoint x="910" y="177" />
         <di:waypoint x="940" y="177" />
       </bpmndi:BPMNEdge>
-      <bpmndi:BPMNEdge id="Flow_Calculate_To_Review_di" bpmnElement="Flow_Calculate_To_Review">
+      <bpmndi:BPMNEdge id="Flow_CallHousehold_To_Summary_di" bpmnElement="Flow_CallHousehold_To_Summary">
         <di:waypoint x="1040" y="177" />
         <di:waypoint x="1070" y="177" />
       </bpmndi:BPMNEdge>
-      <bpmndi:BPMNEdge id="Flow_Review_To_Complete_di" bpmnElement="Flow_Review_To_Complete">
+      <bpmndi:BPMNEdge id="Flow_Summary_To_Calculate_di" bpmnElement="Flow_Summary_To_Calculate">
         <di:waypoint x="1170" y="177" />
         <di:waypoint x="1200" y="177" />
       </bpmndi:BPMNEdge>
-      <bpmndi:BPMNEdge id="Flow_Complete_To_StoreZgwSummary_di" bpmnElement="Flow_Complete_To_StoreZgwSummary">
+      <bpmndi:BPMNEdge id="Flow_Calculate_To_SetAssignee_di" bpmnElement="Flow_Calculate_To_SetAssignee">
         <di:waypoint x="1300" y="177" />
         <di:waypoint x="1330" y="177" />
       </bpmndi:BPMNEdge>
-      <bpmndi:BPMNEdge id="Flow_StoreZgwSummary_To_End_di" bpmnElement="Flow_StoreZgwSummary_To_End">
+      <bpmndi:BPMNEdge id="Flow_SetAssignee_To_SetStatusReview_di" bpmnElement="Flow_SetAssignee_To_SetStatusReview">
         <di:waypoint x="1430" y="177" />
-        <di:waypoint x="1492" y="177" />
+        <di:waypoint x="1460" y="177" />
+      </bpmndi:BPMNEdge>
+      <bpmndi:BPMNEdge id="Flow_SetStatusReview_To_Review_di" bpmnElement="Flow_SetStatusReview_To_Review">
+        <di:waypoint x="1560" y="177" />
+        <di:waypoint x="1590" y="177" />
+      </bpmndi:BPMNEdge>
+      <bpmndi:BPMNEdge id="Flow_Review_To_RemoveTagVerificatie_di" bpmnElement="Flow_Review_To_RemoveTagVerificatie">
+        <di:waypoint x="1690" y="177" />
+        <di:waypoint x="1720" y="177" />
+      </bpmndi:BPMNEdge>
+      <bpmndi:BPMNEdge id="Flow_RemoveTagVerificatie_To_AddTagGoedgekeurd_di" bpmnElement="Flow_RemoveTagVerificatie_To_AddTagGoedgekeurd">
+        <di:waypoint x="1820" y="177" />
+        <di:waypoint x="1850" y="177" />
+      </bpmndi:BPMNEdge>
+      <bpmndi:BPMNEdge id="Flow_AddTagGoedgekeurd_To_Unassign_di" bpmnElement="Flow_AddTagGoedgekeurd_To_Unassign">
+        <di:waypoint x="1950" y="177" />
+        <di:waypoint x="1980" y="177" />
+      </bpmndi:BPMNEdge>
+      <bpmndi:BPMNEdge id="Flow_Unassign_To_SetStatusAfgehandeld_di" bpmnElement="Flow_Unassign_To_SetStatusAfgehandeld">
+        <di:waypoint x="2080" y="177" />
+        <di:waypoint x="2110" y="177" />
+      </bpmndi:BPMNEdge>
+      <bpmndi:BPMNEdge id="Flow_SetStatusAfgehandeld_To_Complete_di" bpmnElement="Flow_SetStatusAfgehandeld_To_Complete">
+        <di:waypoint x="2210" y="177" />
+        <di:waypoint x="2240" y="177" />
+      </bpmndi:BPMNEdge>
+      <bpmndi:BPMNEdge id="Flow_Complete_To_StoreZgwSummary_di" bpmnElement="Flow_Complete_To_StoreZgwSummary">
+        <di:waypoint x="2340" y="177" />
+        <di:waypoint x="2500" y="177" />
+      </bpmndi:BPMNEdge>
+      <bpmndi:BPMNEdge id="Flow_StoreZgwSummary_To_End_di" bpmnElement="Flow_StoreZgwSummary_To_End">
+        <di:waypoint x="2600" y="177" />
+        <di:waypoint x="2662" y="177" />
       </bpmndi:BPMNEdge>
     </bpmndi:BPMNPlane>
   </bpmndi:BPMNDiagram>

--- a/backend/app/gzac/src/dev/resources/config/case/energy-subsidy-request/1-0-0/case/internal-status/energy-subsidy-request.internal-case-status.json
+++ b/backend/app/gzac/src/dev/resources/config/case/energy-subsidy-request/1-0-0/case/internal-status/energy-subsidy-request.internal-case-status.json
@@ -1,0 +1,44 @@
+[
+    {
+        "key": "aanvraag-ontvangen",
+        "title": "Aanvraag ontvangen",
+        "retentionPeriodInDays": -1,
+        "visibleInCaseListByDefault": true,
+        "color": "BLUE"
+    },
+    {
+        "key": "berekening-gestart",
+        "title": "Berekening gestart",
+        "retentionPeriodInDays": -1,
+        "visibleInCaseListByDefault": true,
+        "color": "PURPLE"
+    },
+    {
+        "key": "verificatie-gestart",
+        "title": "Verificatie gestart",
+        "retentionPeriodInDays": -1,
+        "visibleInCaseListByDefault": true,
+        "color": "MAGENTA"
+    },
+    {
+        "key": "inkomenstoets-gestart",
+        "title": "Inkomenstoets gestart",
+        "retentionPeriodInDays": -1,
+        "visibleInCaseListByDefault": true,
+        "color": "CYAN"
+    },
+    {
+        "key": "in-beoordeling",
+        "title": "In beoordeling",
+        "retentionPeriodInDays": -1,
+        "visibleInCaseListByDefault": true,
+        "color": "TEAL"
+    },
+    {
+        "key": "afgehandeld",
+        "title": "Afgehandeld",
+        "retentionPeriodInDays": 365,
+        "visibleInCaseListByDefault": true,
+        "color": "GREEN"
+    }
+]

--- a/backend/app/gzac/src/dev/resources/config/case/energy-subsidy-request/1-0-0/case/tag/energy-subsidy-request.case-tag.json
+++ b/backend/app/gzac/src/dev/resources/config/case/energy-subsidy-request/1-0-0/case/tag/energy-subsidy-request.case-tag.json
@@ -1,0 +1,17 @@
+[
+    {
+        "key": "prioriteit-hoog",
+        "title": "Prioriteit hoog",
+        "color": "RED"
+    },
+    {
+        "key": "verificatie-vereist",
+        "title": "Verificatie vereist",
+        "color": "WARMGRAY"
+    },
+    {
+        "key": "goedgekeurd",
+        "title": "Goedgekeurd",
+        "color": "GREEN"
+    }
+]

--- a/backend/process-document/src/main/kotlin/com/ritense/processdocument/autoconfigure/ProcessDocumentsAutoConfiguration.kt
+++ b/backend/process-document/src/main/kotlin/com/ritense/processdocument/autoconfigure/ProcessDocumentsAutoConfiguration.kt
@@ -91,12 +91,14 @@ class ProcessDocumentsAutoConfiguration {
     fun documentDelegate(
         processDocumentService: ProcessDocumentService,
         userManagementService: UserManagementService,
-        documentService: DocumentService
+        documentService: DocumentService,
+        caseDocumentResolver: CaseDocumentResolver,
     ): DocumentDelegate {
         return DocumentDelegate(
             processDocumentService,
             userManagementService,
-            documentService
+            documentService,
+            caseDocumentResolver,
         )
     }
 
@@ -120,6 +122,7 @@ class ProcessDocumentsAutoConfiguration {
         jsonSchemaDocumentService: JsonSchemaDocumentService,
         userManagementService: UserManagementService,
         objectMapper: ObjectMapper,
+        caseDocumentResolver: CaseDocumentResolver,
     ): DocumentDelegateService {
         return DocumentDelegateService(
             processDocumentService,
@@ -127,6 +130,7 @@ class ProcessDocumentsAutoConfiguration {
             jsonSchemaDocumentService,
             userManagementService,
             objectMapper,
+            caseDocumentResolver,
         )
     }
 

--- a/backend/process-document/src/main/kotlin/com/ritense/processdocument/domain/impl/delegate/DocumentDelegate.kt
+++ b/backend/process-document/src/main/kotlin/com/ritense/processdocument/domain/impl/delegate/DocumentDelegate.kt
@@ -21,6 +21,7 @@ import com.ritense.document.service.DocumentService
 import com.ritense.processdocument.domain.impl.OperatonProcessInstanceId
 import com.ritense.processdocument.service.ProcessDocumentService
 import com.ritense.valtimo.contract.authentication.UserManagementService
+import com.ritense.valtimo.contract.document.CaseDocumentResolver
 import io.github.oshai.kotlinlogging.KotlinLogging
 import org.operaton.bpm.engine.delegate.DelegateExecution
 
@@ -29,6 +30,7 @@ class DocumentDelegate(
     val processDocumentService: ProcessDocumentService,
     val userManagementService: UserManagementService,
     val documentService: DocumentService,
+    val caseDocumentResolver: CaseDocumentResolver,
 ) {
 
     fun setAssignee(execution: DelegateExecution, userEmail: String?) {
@@ -37,18 +39,23 @@ class DocumentDelegate(
                 unassign(execution)
             }
             logger.debug("Assigning user {} to document {}", userEmail, execution.processBusinessKey)
-            val documentId = processDocumentService.getDocumentId(OperatonProcessInstanceId(execution.processInstanceId), execution)
+            val caseDocumentId = getCaseDocumentId(execution)
             val user = userManagementService.findByEmail(userEmail)
                 .orElseThrow { IllegalArgumentException("No user found with email: $userEmail") }
             AuthorizationContext
-                .runWithoutAuthorization { documentService.assignUserToDocument(documentId.id, user.id) }
+                .runWithoutAuthorization { documentService.assignUserToDocument(caseDocumentId, user.id) }
         }
     }
 
     fun unassign(execution: DelegateExecution) {
         logger.debug("Unassigning user from document {}", execution.processBusinessKey)
+        val caseDocumentId = getCaseDocumentId(execution)
+        documentService.unassignUserFromDocument(caseDocumentId)
+    }
+
+    private fun getCaseDocumentId(execution: DelegateExecution): java.util.UUID {
         val documentId = processDocumentService.getDocumentId(OperatonProcessInstanceId(execution.processInstanceId), execution)
-        documentService.unassignUserFromDocument(documentId.id)
+        return caseDocumentResolver.resolveCaseDocumentId(documentId.id)
     }
 
     companion object {

--- a/backend/process-document/src/main/kotlin/com/ritense/processdocument/service/DocumentDelegateService.kt
+++ b/backend/process-document/src/main/kotlin/com/ritense/processdocument/service/DocumentDelegateService.kt
@@ -28,6 +28,7 @@ import com.ritense.document.service.impl.JsonSchemaDocumentService
 import com.ritense.processdocument.domain.impl.OperatonProcessInstanceId
 import com.ritense.valtimo.contract.annotation.SkipComponentScan
 import com.ritense.valtimo.contract.authentication.UserManagementService
+import com.ritense.valtimo.contract.document.CaseDocumentResolver
 import io.github.oshai.kotlinlogging.KotlinLogging
 import org.operaton.bpm.engine.delegate.DelegateExecution
 import org.springframework.stereotype.Service
@@ -44,6 +45,7 @@ class DocumentDelegateService(
     private val jsonSchemaDocumentService: JsonSchemaDocumentService,
     private val userManagementService: UserManagementService,
     private val objectMapper: ObjectMapper,
+    private val caseDocumentResolver: CaseDocumentResolver,
 ) {
 
     fun getDocumentVersion(execution: DelegateExecution): Int? {
@@ -96,40 +98,41 @@ class DocumentDelegateService(
         }
         logger.debug("Assigning user {} to document {}", userEmail, execution.processBusinessKey)
 
-        val processInstanceId = OperatonProcessInstanceId(execution.processInstanceId)
-        val documentId = processDocumentService.getDocumentId(processInstanceId, execution)
+        val caseDocumentId = getCaseDocumentId(execution)
         val user = runWithoutAuthorization { userManagementService.findByEmail(userEmail) }
             .orElseThrow { IllegalArgumentException("No user found with email: $userEmail") }
-        documentService.assignUserToDocument(documentId.id, user.id)
+        documentService.assignUserToDocument(caseDocumentId, user.id)
     }
 
     fun setInternalStatus(execution: DelegateExecution, statusKey: String?) {
-        val processInstanceId = OperatonProcessInstanceId(execution.processInstanceId)
-        val documentId = processDocumentService.getDocumentId(processInstanceId, execution)
+        val caseDocumentId = getCaseDocumentId(execution)
 
-        documentService.setInternalStatus(documentId, statusKey)
+        documentService.setInternalStatus(JsonSchemaDocumentId.existingId(caseDocumentId), statusKey)
     }
 
     fun addCaseTag(execution: DelegateExecution, caseTagKey: String) {
-        val processInstanceId = OperatonProcessInstanceId(execution.processInstanceId)
-        val documentId = processDocumentService.getDocumentId(processInstanceId, execution)
+        val caseDocumentId = getCaseDocumentId(execution)
 
-        documentService.addCaseTag(documentId, caseTagKey)
+        documentService.addCaseTag(JsonSchemaDocumentId.existingId(caseDocumentId), caseTagKey)
     }
 
     fun removeCaseTag(execution: DelegateExecution, caseTagKey: String) {
-        val processInstanceId = OperatonProcessInstanceId(execution.processInstanceId)
-        val documentId = processDocumentService.getDocumentId(processInstanceId, execution)
+        val caseDocumentId = getCaseDocumentId(execution)
 
-        documentService.removeCaseTag(documentId, caseTagKey)
+        documentService.removeCaseTag(JsonSchemaDocumentId.existingId(caseDocumentId), caseTagKey)
     }
 
     fun unassign(execution: DelegateExecution) {
         logger.debug("Unassigning user from document {}", execution.processBusinessKey)
 
+        val caseDocumentId = getCaseDocumentId(execution)
+        documentService.unassignUserFromDocument(caseDocumentId)
+    }
+
+    private fun getCaseDocumentId(execution: DelegateExecution): UUID {
         val processInstanceId = OperatonProcessInstanceId(execution.processInstanceId)
         val documentId = processDocumentService.getDocumentId(processInstanceId, execution)
-        documentService.unassignUserFromDocument(documentId.id)
+        return caseDocumentResolver.resolveCaseDocumentId(documentId.id)
     }
 
     private fun findOptionalValueByJsonPointer(jsonPointer: String?, execution: DelegateExecution): Optional<Any> {

--- a/backend/process-document/src/test/kotlin/com/ritense/processdocument/domain/impl/delegate/DocumentDelegateTest.kt
+++ b/backend/process-document/src/test/kotlin/com/ritense/processdocument/domain/impl/delegate/DocumentDelegateTest.kt
@@ -24,8 +24,10 @@ import com.ritense.valtimo.contract.OauthConfigHolder
 import com.ritense.valtimo.contract.authentication.UserManagementService
 import com.ritense.valtimo.contract.authentication.model.ValtimoUserBuilder
 import com.ritense.valtimo.contract.config.ValtimoProperties.Oauth
+import com.ritense.valtimo.contract.document.CaseDocumentResolver
 import org.junit.jupiter.api.BeforeEach
 import org.junit.jupiter.api.Test
+import org.mockito.kotlin.any
 import org.mockito.kotlin.mock
 import org.mockito.kotlin.times
 import org.mockito.kotlin.verify
@@ -39,6 +41,7 @@ internal class DocumentDelegateTest {
     lateinit var processDocumentService: ProcessDocumentService
     lateinit var userManagementService: UserManagementService
     lateinit var documentService: DocumentService
+    lateinit var caseDocumentResolver: CaseDocumentResolver
     lateinit var documentDelegate: DocumentDelegate
 
     @BeforeEach
@@ -46,10 +49,13 @@ internal class DocumentDelegateTest {
         processDocumentService = mock()
         userManagementService = mock()
         documentService = mock()
+        caseDocumentResolver = mock()
+        whenever(caseDocumentResolver.resolveCaseDocumentId(any())).thenAnswer { it.arguments[0] }
         documentDelegate = DocumentDelegate(
             processDocumentService,
             userManagementService,
             documentService,
+            caseDocumentResolver,
         )
         OauthConfigHolder(Oauth())
     }

--- a/backend/process-document/src/test/kotlin/com/ritense/processdocument/service/DocumentDelegateServiceTest.kt
+++ b/backend/process-document/src/test/kotlin/com/ritense/processdocument/service/DocumentDelegateServiceTest.kt
@@ -28,6 +28,7 @@ import com.ritense.valtimo.contract.OauthConfigHolder
 import com.ritense.valtimo.contract.authentication.UserManagementService
 import com.ritense.valtimo.contract.authentication.model.ValtimoUserBuilder
 import com.ritense.valtimo.contract.config.ValtimoProperties.Oauth
+import com.ritense.valtimo.contract.document.CaseDocumentResolver
 import com.ritense.valtimo.contract.json.MapperSingleton
 import org.junit.jupiter.api.Assertions.assertEquals
 import org.junit.jupiter.api.BeforeEach
@@ -37,6 +38,7 @@ import org.mockito.kotlin.mock
 import org.mockito.kotlin.times
 import org.mockito.kotlin.verify
 import org.mockito.kotlin.whenever
+import org.mockito.kotlin.any as anyKt
 import org.operaton.bpm.engine.delegate.DelegateExecution
 import java.time.LocalDateTime
 import java.util.Optional
@@ -49,6 +51,7 @@ internal class DocumentDelegateServiceTest : BaseTest() {
     private lateinit var documentService: DocumentService
     private lateinit var jsonSchemaDocumentService: JsonSchemaDocumentService
     private lateinit var userManagementService: UserManagementService
+    private lateinit var caseDocumentResolver: CaseDocumentResolver
     private lateinit var documentDelegateService: DocumentDelegateService
 
     lateinit var definition: JsonSchemaDocumentDefinition
@@ -76,12 +79,15 @@ internal class DocumentDelegateServiceTest : BaseTest() {
         documentService = mock()
         jsonSchemaDocumentService = mock()
         jsonSchemaDocumentService = mock()
+        caseDocumentResolver = mock()
+        whenever(caseDocumentResolver.resolveCaseDocumentId(anyKt())).thenAnswer { it.arguments[0] }
         documentDelegateService = DocumentDelegateService(
             processDocumentService,
             documentService,
             jsonSchemaDocumentService,
             userManagementService,
-            MapperSingleton.get()
+            MapperSingleton.get(),
+            caseDocumentResolver
         )
         delegateExecution = mock<DelegateExecution>()
         whenever(delegateExecution.id).thenReturn("id")
@@ -274,12 +280,14 @@ internal class DocumentDelegateServiceTest : BaseTest() {
 
         documentDelegateService.setAssignee(delegateExecution, "john@example.com")
 
+        verify(caseDocumentResolver).resolveCaseDocumentId(UUID.fromString(documentId))
         verify(documentService, times(1)).assignUserToDocument(UUID.fromString(documentId), "anId")
     }
 
     @Test
     fun `should set status to document`() {
-        val documentId = JsonSchemaDocumentId.existingId(UUID.fromString("11111111-1111-1111-1111-111111111111"))
+        val documentUuid = UUID.fromString("11111111-1111-1111-1111-111111111111")
+        val documentId = JsonSchemaDocumentId.existingId(documentUuid)
         val processInstanceId = "00000000-0000-0000-0000-000000000000"
         val delegateExecution = mock<DelegateExecution>()
         whenever(delegateExecution.id).thenReturn("id")
@@ -292,12 +300,37 @@ internal class DocumentDelegateServiceTest : BaseTest() {
         val newStatus = "test"
         documentDelegateService.setInternalStatus(delegateExecution, newStatus)
 
+        verify(caseDocumentResolver).resolveCaseDocumentId(documentUuid)
         verify(documentService).setInternalStatus(documentId, newStatus)
     }
 
     @Test
+    fun `should set status to parent case document when called from building block`() {
+        val buildingBlockDocumentUuid = UUID.fromString("11111111-1111-1111-1111-111111111111")
+        val caseDocumentUuid = UUID.fromString("22222222-2222-2222-2222-222222222222")
+        val buildingBlockDocumentId = JsonSchemaDocumentId.existingId(buildingBlockDocumentUuid)
+        val caseDocumentId = JsonSchemaDocumentId.existingId(caseDocumentUuid)
+        val processInstanceId = "00000000-0000-0000-0000-000000000000"
+        val delegateExecution = mock<DelegateExecution>()
+        whenever(delegateExecution.id).thenReturn("id")
+        whenever(delegateExecution.processInstanceId).thenReturn(processInstanceId)
+        whenever(delegateExecution.processBusinessKey).thenReturn(buildingBlockDocumentUuid.toString())
+        whenever(
+            processDocumentService.getDocumentId(OperatonProcessInstanceId(processInstanceId), delegateExecution)
+        ).thenReturn(buildingBlockDocumentId)
+        whenever(caseDocumentResolver.resolveCaseDocumentId(buildingBlockDocumentUuid)).thenReturn(caseDocumentUuid)
+
+        val newStatus = "test"
+        documentDelegateService.setInternalStatus(delegateExecution, newStatus)
+
+        verify(caseDocumentResolver).resolveCaseDocumentId(buildingBlockDocumentUuid)
+        verify(documentService).setInternalStatus(caseDocumentId, newStatus)
+    }
+
+    @Test
     fun `should add case tag to document`() {
-        val documentId = JsonSchemaDocumentId.existingId(UUID.fromString("11111111-1111-1111-1111-111111111111"))
+        val documentUuid = UUID.fromString("11111111-1111-1111-1111-111111111111")
+        val documentId = JsonSchemaDocumentId.existingId(documentUuid)
         val processInstanceId = "00000000-0000-0000-0000-000000000000"
         val delegateExecution = mock<DelegateExecution>()
         whenever(delegateExecution.id).thenReturn("id")
@@ -309,12 +342,14 @@ internal class DocumentDelegateServiceTest : BaseTest() {
 
         documentDelegateService.addCaseTag(delegateExecution, "important")
 
+        verify(caseDocumentResolver).resolveCaseDocumentId(documentUuid)
         verify(documentService).addCaseTag(documentId, "important")
     }
 
     @Test
     fun `should remove case tag from document`() {
-        val documentId = JsonSchemaDocumentId.existingId(UUID.fromString("11111111-1111-1111-1111-111111111111"))
+        val documentUuid = UUID.fromString("11111111-1111-1111-1111-111111111111")
+        val documentId = JsonSchemaDocumentId.existingId(documentUuid)
         val processInstanceId = "00000000-0000-0000-0000-000000000000"
         val delegateExecution = mock<DelegateExecution>()
         whenever(delegateExecution.id).thenReturn("id")
@@ -326,6 +361,7 @@ internal class DocumentDelegateServiceTest : BaseTest() {
 
         documentDelegateService.removeCaseTag(delegateExecution, "important")
 
+        verify(caseDocumentResolver).resolveCaseDocumentId(documentUuid)
         verify(documentService).removeCaseTag(documentId, "important")
     }
 
@@ -343,6 +379,7 @@ internal class DocumentDelegateServiceTest : BaseTest() {
 
         documentDelegateService.unassign(delegateExecution)
 
+        verify(caseDocumentResolver).resolveCaseDocumentId(UUID.fromString(documentId))
         verify(documentService, times(1)).unassignUserFromDocument(UUID.fromString(documentId))
     }
 

--- a/documentation/release-notes/13.x.x/13.21.0/README.md
+++ b/documentation/release-notes/13.x.x/13.21.0/README.md
@@ -18,4 +18,5 @@
 
 ## Bugfixes
 
-* New bugfix.
+* Fixed a bug where building blocks could not update the internal case status, case tags, or assignee of the calling
+  case. The building block now correctly references the case definition of the parent case.

